PhysicsTools/PatAlgos/src/MuonMvaEstimator.cc

```diff
@@ -0,0 +1,155 @@
+  pt_                   = muon.pt();
+  eta_                  = muon.eta();
+  segmentCompatibility_ = muon.segmentCompatibility();
+  miniRelIsoCharged_ = muon.miniPFIsolation().chargedHadronIso();
+  miniRelIsoNeutral_ = muon.miniPFIsolation().neutralHadronIso();
+
+  double dB2D  = muon.dB(pat::Muon::BS2D);
+  double dB3D  = muon.dB(pat::Muon::PV3D);
+  double edB3D = muon.edB(pat::Muon::PV3D);
+  sip_  = edB3D>0?fabs(dB3D/edB3D):0.0; 
+  log_abs_dxyBS_     = log(fabs(dB2D)); 
+  log_abs_dzPV_      = log(fabs(muon.muonBestTrack()->dz(vertex.position())));
+
+  // compute jet-related information
+  // jetPtRel_          = jetPtRel;
+  // jetPtRatio_        = std::min(jetPtRatio,1.5);
+  // jetBTagCSV_        = std::max(jetBTagCSV,0.0);
+  // jetNDauCharged_    = jetNDauCharged;
+
+  //Initialise loop variables
+  double minDr = 9999;
+  double jecL1L2L3Res = 1.;
+  double jecL1 = 1.;
+
+  jetPtRatio_ = -99;
+  jetPtRel_   = -99;
+  jetBTagCSV_ = -999;
+  jetNDauCharged_ = -1;
+
+  for (const auto& tagI: bTags){
+    // for each muon with the lepton 
+    double dr = deltaR(*(tagI.first), muon);
+    if(dr > minDr) continue;  
+    minDr = dr;
+      
+    reco::Candidate::LorentzVector muP4(muon.p4()); 
+    reco::Candidate::LorentzVector jetP4(tagI.first->p4());
+
+    if (correctorL1 && correctorL1L2L3Res){
+      jecL1L2L3Res = correctorL1L2L3Res->correction(*(tagI.first));
+      jecL1 = correctorL1->correction(*(tagI.first));
+    }
+
+    // Get b-jet info
+    jetBTagCSV_ = tagI.second;
+    jetNDauCharged_ = 0;
+    for (auto jet: tagI.first->getJetConstituentsQuick()){
+      const reco::PFCandidate *pfcand = dynamic_cast<const reco::PFCandidate*>(jet);
+      if (pfcand==0) continue;
+      if (pfcand->charge()==0) continue;
+      if (!pfcand->bestTrack()) continue;
+      if (!pfcand->bestTrack()->quality(reco::Track::highPurity)) continue;
+      if (pfcand->bestTrack()->pt()<1.) continue;
+      if (pfcand->bestTrack()->hitPattern().numberOfValidHits()<8) continue;
+      if (pfcand->bestTrack()->hitPattern().numberOfValidPixelHits()<2) continue;
+      if (pfcand->bestTrack()->normalizedChi2()>=5) continue;
+
+      if (std::fabs(pfcand->bestTrack()->dxy(vertex.position())) > 0.2) continue;
+      if (std::fabs(pfcand->bestTrack()->dz(vertex.position())) > 17) continue;
+      jetNDauCharged_++;
+    }
+
+    if(minDr < dRmax_){
+      if ((jetP4-muP4).Rho()<0.0001){ 
+	jetPtRel_ = 0;
+	jetPtRatio_ = 1;
+      } else {
+	jetP4 -= muP4/jecL1;
+	jetP4 *= jecL1L2L3Res;
+	jetP4 += muP4;
+      
+	jetPtRatio_ = muP4.pt()/jetP4.pt();
+	jetPtRel_ = ptRel(muP4,jetP4);
+      }
+    }
+  }
+
+  if (jetPtRatio_>1.5) jetPtRatio_ = 1.5;
+  if (jetBTagCSV_<0) jetBTagCSV_ = 0;
+  mva_ = tmvaReader_.EvaluateMVA("BDTG");
+  // if (pt_>20){
+  //   printf("pt: %0.6f\n",pt_);
+  //   printf("\teta:\t %0.6f\n",eta_);
+  //   printf("\tjetNDauCharged:\t %0.6f\n",jetNDauCharged_);
```

Comment:
is this commented out code needed?
Please remove or add comments inline in the code why the commented out block is relevant

Reply:
fixed